pkg/iac-providers/terraform/v12/load-dir.go

```diff
@@ -84,9 +83,7 @@ func (*TfV12) LoadIacDir(absRootDir string) (allResourcesConfig output.AllResour
 			if isLocalSourceAddr(req.SourceAddr) {
 				// determine the absolute path from root module to the sub module
 				// using *configs.ModuleRequest.Path field
-				pathArr := strings.Split(req.Path.String(), ".")
-				pathArr = pathArr[:len(pathArr)-1]
-				pathToModule = filepath.Join(absRootDir, filepath.Join(pathArr...), req.SourceAddr)
+				pathToModule = filepath.Join(absRootDir, req.Parent.SourceAddr, req.SourceAddr)
```

Comment:
I'm concerned that this only moves the issue one level deeper.  If you go back to https://github.com/salexpdx/terrascan/commit/89b06f6b18881274ff7175bf418cf7a463077fa0#diff-ed0a902412e1c2c9cca7b9c8d22f14edcd3711f1535d0bc128e801e3ac300c84R95, you can see that i looped through.  That is done so that if there is more than one layer of sub-modules, it will parse through all the way.